text/66-self-managed-stack-sets.md

```diff
@@ -0,0 +1,278 @@
+The intermdiate stack may not be ideal for some customers, and may require large migration effort for customers who manage many stacksets.
+
+### What changes are required to enable this change?
+
+To enable this feature, we would need to introduce:
+
+* A StackSet construct for customers to extend
+* A new assembly/artifact type for the Cloud Assembly
+* A new stack set synthesizer
+* Deployment code that creates/updates stack sets using AWS CloudFormation APIs.
+
+### Is this a breaking change?
+
+No.
+
+### What are the drawbacks of this solution?
+
+The handling of CDK assets is more complex than a normal stack CDK app.
+Stack set instances are deployed independently from CDK context, and exist in different accounts/regions.
+This is dicussed more in the design details.
+
+### What alternative solutions did you consider?
+
+An alternative is the usage of `cdk.CfnStackSet`, which is not ideal for customers with existing stack sets since there is significant migration effort.
+
+### What is the high level implementation plan?
+
+* Update default boostrap template to include default stack set roles
+* Create StackSet core construct
+* Create StackSet artifact type in the Cloud Assembly
+* Create StackSet Synthesizer
+* Update CLI to collect StackSet artifact types in addition to Stacks
+* Add deployment code for creating/updating stack sets
+
+### Are there any open issues that need to be addressed later?
+
+* CDK asset handling
+* Future support for service managed stack sets
+* Future support for delete stack set
+* CDK Pipelines
+
+## Appendix
+
+### CDK Bootstrap Template Addition
+
+To support stack sets,
+a stack set administration role should be created in the parent account and a stack set execution role should be created in the child accounts.
+To make this process easier, default roles have been added to the default bootstrap template. (Bootstrapping steps describe in the README section).
+
+```yaml
+  # ...
+  CloudFormationStackSetAdministrationRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Statement:
+          - Action: sts:AssumeRole
+            Effect: Allow
+            Principal:
+              Service: cloudformation.amazonaws.com
+        Version: '2012-10-17'
+      RoleName:
+        Fn::Sub: cdk-${Qualifier}-stack-set-admin-role-${AWS::AccountId}-${AWS::Region}
+  CloudFormationStackSetAdministrationRolePolicy:
+    Type: AWS::IAM::Policy
+    Properties:
+      PolicyDocument:
+        Statement:
+          - Action:
+              - sts:AssumeRole
+            Resource:
+              - Fn::Sub: "arn:aws:iam::*:role/cdk-${Qualifier}-stack-set-exec-role-${AWS::Region}"
+            Effect: Allow
+        Version: '2012-10-17'
+      Roles:
+        - Ref: CloudFormationStackSetAdministrationRole
+      PolicyName:
+        Fn::Sub: cdk-${Qualifier}-stack-set-admin-role-policy-${AWS::Region}
```

Comment:
Can we switch this to an inline policy on the `CloudFormationStackSetAdministrationRole`?

Reply:
Yep, will do -- was just following the pattern from the other IAM role resources